README.md

```diff
@@ -97,11 +103,162 @@ Please see Hazelcast Node.js [code samples](https://github.com/hazelcast/hazelca
+and how you should set paths and exported names for the client to load objects.
+
+## 1. Configuration Options
 
 You can configure Hazelcast Node.js Client declaratively (JSON) or programmatically (API).
 
-See [CONFIG.md](CONFIG.md) for details.
+* Programmatic configuration
+* Declarative configuration (JSON file)
+
+### 1.1. Programmatic Configuration
+
+For programmatic configuration of the Hazelcast Node.js Client, just instantiate a `ClientConfig` object and configure the
+desired aspects. An example is shown below.
+
+```javascript
+var Config = require('hazelcast-client').Config;
+var Address = require('hazelcast-client').Address;
+var cfg = new Config.ClientConfig();
+cfg.networkConfig.addresses.push('127.0.0.1:5701');
+return HazelcastClient.newHazelcastClient(cfg);
+```
+
+Refer to `ClientConfig` class documentation at [Hazelcast Node.js Client API Docs](http://hazelcast.github.io/hazelcast-nodejs-client/api/current/docs) for details.
+
+### 1.2. Declarative Configuration (JSON)
+
+For declarative configuration, the client looks at the following places for the client configuration file.
+
+1. Environment variable: The client first looks for the environment variable `HAZELCAST_CLIENT_CONFIG`. If it exists,
+the client looks for the configuration file in the specified location.
+2. Current working directory: If there is no environment variable set, the client tries to load `hazelcast-client.json`
+from the current working directory.
+3. Default configuration: If all of the above methods fail, the client starts with the default configuration.
+The default configuration is programmatic. If you want to override the default configuration declaratively, you need to create
+a `hazelcast-client.json` file in your working directory. If you want to have an example for this file, you can find `hazelcast-client-default.json`
+and `hazelcast-client-sample.json` files in the Github repository.
+
+Following is a sample JSON configuration file:
+
+```json
+{
+    "group": {
+        "name": "hazel",
+        "password": "cast"
+    },
+    "properties": {
+        "hazelcast.client.heartbeat.timeout": 10000,
+        "hazelcast.client.invocation.retry.pause.millis": 4000,
+        "hazelcast.client.invocation.timeout.millis": 180000,
+        "hazelcast.invalidation.reconciliation.interval.seconds": 50,
+        "hazelcast.invalidation.max.tolerated.miss.count": 15,
+        "hazelcast.invalidation.min.reconciliation.interval.seconds": 60
+    },
+    "network": {
+        "clusterMembers": [
+            "127.0.0.1:5701"
+        ],
+        "smartRouting": true,
+        "connectionTimeout": 6000,
+        "connectionAttemptPeriod": 4000,
+        "connectionAttemptLimit": 3
+    }
+}
+```
+
+In the following chapters you will learn the description of all elements included in a JSON configuration file used to configure Hazelcast Node.js client.
+
+## 2. Importing Multiple Configurations
```

Comment:
Are we sure that this section is an `Guide to using the Node.js Client with Hazelcast` section? It feels a bit unneccessary for someone trying to get started with Node.js client. If we are going to keep `CONFIG.md` in the end, this should be only there I think. @Serdaro

Reply:
@mustafaiman , What I think is: since this "composing a configuration"  is a Hazelcast feature, it would look nice to mention it with examples (as it is now) here, so that the user can learn how the importing mechanism works when a Node.js client is being configured. And since I think we agreed on moving the CONFIG.md content to the main README, there is no harm keeping this here.

Reply:
sure 👍